extensions/panache/hibernate-orm-panache/deployment/src/main/java/io/quarkus/hibernate/orm/panache/deployment/PanacheResourceProcessor.java

```diff
@@ -78,6 +81,16 @@ void build(CombinedIndexBuildItem index,
             BuildProducer<PanacheEntityClassesBuildItem> entityClasses,
             BuildProducer<NamedQueryEntityClassBuildStep> namedQueries) throws Exception {
 
+        // we verify that no ID fields are defined (via @BsonId) when extending PanacheMongoEntity
+        for (AnnotationInstance annotationInstance : index.getIndex().getAnnotations(DOTNAME_ID)) {
+            ClassInfo info = io.quarkus.panache.common.deployment.JandexUtil.getEnclosingClass(annotationInstance);
+            if (info.superName().equals(DOTNAME_PANACHE_ENTITY)) {
```

Comment:
This should be recursive: it is possible that we have superclasses between the entity class and `PanacheEntity`.

Reply:
Done, I climb up the hierarchy of classes.